compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -806,9 +802,18 @@ export class Environment {
            * `import {useHook as foo} ...`
            * `import {foo as useHook} ...`
            */
-          return isHookName(binding.imported) || isHookName(binding.name)
-            ? this.#getCustomHookType()
-            : null;
+          const expectHook =
+            isHookName(binding.imported) || isHookName(binding.name);
+          if (expectHook) {
+            if (
+              propertyType &&
+              getHookKindForType(this, propertyType) !== null
+            ) {
+              return propertyType;
+            }
+            return this.#getCustomHookType();
```

Comment:
We might produce invalid compilation output on false positives (i.e. functions that are hook-named but not hooks) as hook calls freeze args and returned results whereas regular function calls don't. Imports being renamed nonhook <--> hook look a bit suspicious, but we haven't had any practical issues.
```js
import {foo as useHookNameForNonHook} ...
```

Seems reasonable as our internal use is currently pretty limited. It might be a good idea to warn / bailout here in the future if we find cases of calls being incorrectly inferred as hooks.